### DISCUSSION TITLE
Rule in contrast with mandatory fields ?

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#491] [Q&A] [answered]
Hi Lab !
this rules seem , in my opinion , in contrast with mandatory fields

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/7563c446-6b30-4b40-b297-08134465a4d7)

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/ca137c9a-2413-406d-ab28-fd55423bc894)

I am using sdk 1.7, notice subtype 7
it seems that the 2 fields ( BT-131(t)-Lot and BT-1311(t)-Lot ) should be  in "XOR" condition to obey to the rules
Bye and have a nice weekend
Renato

**Comment by rengit62** [comment]
Hi lab 
I did not explain clear what happens.
The 2 time fields are mandatory but filling only those I get 2 validation error : 
BT-1311(t)-Lot is not allowed in notice type '7' under following condition: Deadline receipt Requests date (BT-1311(d)-Lot) is not present
BT-131(t)-Lot is not allowed in notice type '7' under following condition: Deadline receipt Tenders date (BT-131(d)-Lot) is not present

Clear, if you set the time you have to set the date, also.

Now I set the 2 dates ( that are not mandatory ) : 
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/295d4f87-51fb-474e-9501-2031286e4d73)

call the cvs validation and get : 

BT-131(d)-Lot is not allowed in notice type '7' under following condition: Deadline receipt Requests date (BT-1311(d)-Lot) is present
BT-1311(d)-Lot is not allowed in notice type '7' under following condition: Deadline receipt Tenders date (BT-131(d)-Lot) is present

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/bdb93274-887c-4d1e-bc96-604dde5135c7)


How can I solve this situation ?
Thanks and regards
Renato

**Reply by YvesJo** [reply]
Hi,
The two deadlines are exclusive, so just fill one of them and ensure that for that specific deadline both date and time are provided

**Reply by rengit62** [reply]
It is clear that they are exclusive, my question is , why, if they are exclusive, are both mandatory ?
In notice 18 not only the time (exclusive) fields are mandatory , but the date field also.
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/a9487bdc-d2b1-4905-a959-83bfaf374358)
Would be much simpler to remove mandatory flag and to check that only one of these was filled ?
But not only that , suppose someone fill both because they are mandatory, which of them the app should remove ?
If there was  a rule to check that both cannot be set, the user would remove the date that he wants.

In notice 18 the same situation occurs for BT-70-Lot if OPT-060-Lot is not present
Both are mandatory !
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/8bedd686-ffc3-43ee-85e5-54932a9504a5)

bye
Renato

**Reply by YvesJo** [reply]
Hi,
Both appear as mandatory as initially none is specified. Rules aren't reevaluated after one has been specified. Also Conditionally Mandatory rules aren't displayed differently from purely mandatory rules. For the time being, only a validation may help detect and solve an issue.

**Reply by rengit62** [reply]
Perhaps there is a misunderstanding.
Are you writing about eNotices2 behavior ?
We do not use it, we are building an application based entirely on SDK. 
If a field in fields.json is mandatory we use same indication of Eforms-Notice-Editor. When the user save/send the notice, validation occurs.
First check that the field adhere to definition of fields.json ( forbidden/mandatory/pattern ... ) after it sends the notice to cvs validation.
Anyway I made this test : I removed mandatory condition from both fields in fields.json , I filled only OPT-060-LOT but I got again a validation error on BT-70-Lot
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/4db157df-f4fc-4a98-960b-8f4a4b4d6853)

BT-70-Lot is mandatory in notice type '18' except under following condition: OPT-060-Lot is not present

`<?xml version="1.0" encoding="UTF-8"?>
<ContractNotice xmlns="urn:oasis:names:specification:ubl:schema:xsd:ContractNotice-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:efac="http://data.europa.eu/p27/eforms-ubl-extension-aggregate-components/1" xmlns:efbc="http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1" xmlns:efext="http://data.europa.eu/p27/eforms-ubl-extensions/1" xmlns:ext="urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2">
	<ext:UBLExtensions>
		<ext:UBLExtension>
			<ext:ExtensionContent>
				<efext:EformsExtension>
					<efac:NoticeSubType>
						<cbc:SubTypeCode listName="notice-subtype">18</cbc:SubTypeCode>
					</efac:NoticeSubType>
					<efac:Organizations>
						<efac:Organization>
							<efac:Company>
								<cac:PartyIdentification>
									<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
								</cac:PartyIdentification>
								<cac:PartyName>
									<cbc:Name languageID="ENG">Buyer Organization</cbc:Name>
								</cac:PartyName>
								<cac:PostalAddress>
									<cbc:CityName>PISA</cbc:CityName>
									<cbc:CountrySubentityCode listName="nuts">AT113</cbc:CountrySubentityCode>
									<cac:Country>
										<cbc:IdentificationCode listName="country">BEL</cbc:IdentificationCode>
									</cac:Country>
								</cac:PostalAddress>
								<cac:Contact>
									<cbc:Telephone>05123232</cbc:Telephone>
									<cbc:ElectronicMail>kkk@sdsd.sdsd</cbc:ElectronicMail>
								</cac:Contact>
							</efac:Company>
							<efac:TouchPoint>
								<cac:PartyIdentification>
									<cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID>
								</cac:PartyIdentification>
							</efac:TouchPoint>
						</efac:Organization>
					</efac:Organizations>
				</efext:EformsExtension>
			</ext:ExtensionContent>
		</ext:UBLExtension>
	</ext:UBLExtensions>
	<cbc:UBLVersionID>2.3</cbc:UBLVersionID>
	<cbc:CustomizationID>eforms-sdk-1.7</cbc:CustomizationID>
	<cbc:ID schemeName="notice-id">647cc991-0b2a-4284-814f-d7a049a1d9a3</cbc:ID>
	<cbc:ContractFolderID>c2a0dcd1-4d17-4072-9252-38de16ead921</cbc:ContractFolderID>
	<cbc:IssueDate>2023-05-31+02:00</cbc:IssueDate>
	<cbc:IssueTime>04:41:26+02:00</cbc:IssueTime>
	<cbc:VersionID>01</cbc:VersionID>
	<cbc:RegulatoryDomain>31985R2137</cbc:RegulatoryDomain>
	<cbc:NoticeTypeCode listName="competition">cn-standard</cbc:NoticeTypeCode>
	<cbc:NoticeLanguageCode listName="language">ENG</cbc:NoticeLanguageCode>
	<cac:ContractingParty>
		<cac:Party>
			<cac:PartyIdentification>
				<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
			</cac:PartyIdentification>
		</cac:Party>
	</cac:ContractingParty>
	<cac:TenderingProcess>
		<cbc:ProcedureCode listName="procurement-procedure-type">comp-dial</cbc:ProcedureCode>
	</cac:TenderingProcess>
	<cac:ProcurementProject>
		<cbc:Name languageID="ENG">PROC TITLE</cbc:Name>
		<cbc:Description languageID="ENG">PROC DESCR</cbc:Description>
		<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
		<cac:MainCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03000000</cbc:ItemClassificationCode>
		</cac:MainCommodityClassification>
		<cac:AdditionalCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03114000</cbc:ItemClassificationCode>
		</cac:AdditionalCommodityClassification>
	</cac:ProcurementProject>
	<cac:ProcurementProjectLot>
		<cbc:ID schemeName="Lot">LOT-0001</cbc:ID>
		<cac:TenderingTerms>
			<ext:UBLExtensions>
				<ext:UBLExtension>
					<ext:ExtensionContent>
						<efext:EformsExtension>
							<efac:SelectionCriteria>
								<cbc:CriterionTypeCode listName="selection-criterion">ef-stand</cbc:CriterionTypeCode>
							</efac:SelectionCriteria>
							<efac:SelectionCriteria>
								<cbc:CriterionTypeCode listName="selection-criterion">sui-act</cbc:CriterionTypeCode>
							</efac:SelectionCriteria>
							<efac:SelectionCriteria>
								<cbc:CriterionTypeCode listName="selection-criterion">tp-abil</cbc:CriterionTypeCode>
							</efac:SelectionCriteria>
							<efac:TenderSubcontractingRequirements>
								<efbc:TenderSubcontractingRequirementsCode listName="subcontracting-indication">subc-oblig</efbc:TenderSubcontractingRequirementsCode>
							</efac:TenderSubcontractingRequirements>
						</efext:EformsExtension>
					</ext:ExtensionContent>
				</ext:UBLExtension>
			</ext:UBLExtensions>
			<cac:RequiredFinancialGuarantee>
				<cbc:GuaranteeTypeCode listName="tender-guarantee-required">false</cbc:GuaranteeTypeCode>
			</cac:RequiredFinancialGuarantee>
			<cac:CallForTendersDocumentReference>
				<cbc:ID>PROC-DOC-ID</cbc:ID>
				<cbc:DocumentType>restricted-document</cbc:DocumentType>
				<cac:Attachment>
					<cac:ExternalReference>
						<cbc:URI>http://dfdfdf.fgfg</cbc:URI>
					</cac:ExternalReference>
				</cac:Attachment>
			</cac:CallForTendersDocumentReference>
			<cac:PaymentTerms>
				<cbc:Note languageID="ENG">LA DONZELLETTA VIEN DALLA CAMPAGNA IN SUL CALAR DEL SOLE</cbc:Note>
			</cac:PaymentTerms>
			<cac:TendererQualificationRequest>
				<cbc:CompanyLegalFormCode listName="required">false</cbc:CompanyLegalFormCode>
			</cac:TendererQualificationRequest>
			<cac:TendererQualificationRequest>
				<cac:SpecificTendererRequirement>
					<cbc:TendererRequirementTypeCode listName="reserved-procurement">none</cbc:TendererRequirementTypeCode>
				</cac:SpecificTendererRequirement>
			</cac:TendererQualificationRequest>
			<cac:AllowedSubcontractTerms>
				<cbc:SubcontractingConditionsCode listName="subcontracting-obligation">subc-chng</cbc:SubcontractingConditionsCode>
			</cac:AllowedSubcontractTerms>
			<cac:AllowedSubcontractTerms>
				<cbc:MaximumPercent>40</cbc:MaximumPercent>
				<cbc:MinimumPercent>10</cbc:MinimumPercent>
			</cac:AllowedSubcontractTerms>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="nda">true</cbc:ExecutionRequirementCode>
				<cbc:Description languageID="ENG">GUAI A TE SE SPIFFERI QUALCOSA DI QUESTA OPPORTUNITA' DI BUSINESS</cbc:Description>
			</cac:ContractExecutionRequirement>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="reserved-execution">yes</cbc:ExecutionRequirementCode>
			</cac:ContractExecutionRequirement>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="einvoicing">required</cbc:ExecutionRequirementCode>
			</cac:ContractExecutionRequirement>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="conditions">performance</cbc:ExecutionRequirementCode>
			</cac:ContractExecutionRequirement>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="ecatalog-submission">allowed</cbc:ExecutionRequirementCode>
			</cac:ContractExecutionRequirement>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="esignature-submission">true</cbc:ExecutionRequirementCode>
			</cac:ContractExecutionRequirement>
			<cac:AwardingTerms>
				<cac:AwardingCriterion>
					<cbc:CalculationExpression languageID="ENG">E' una supercazzola con scappellamento a destra</cbc:CalculationExpression>
					<cac:SubordinateAwardingCriterion>
						<ext:UBLExtensions>
							<ext:UBLExtension>
								<ext:ExtensionContent>
									<efext:EformsExtension>
										<efac:AwardCriterionParameter>
											<efbc:ParameterNumeric>6</efbc:ParameterNumeric>
										</efac:AwardCriterionParameter>
										<efac:AwardCriterionParameter>
											<efbc:ParameterNumeric>2</efbc:ParameterNumeric>
										</efac:AwardCriterionParameter>
									</efext:EformsExtension>
								</ext:ExtensionContent>
							</ext:UBLExtension>
						</ext:UBLExtensions>
						<cbc:AwardingCriterionTypeCode listName="award-criterion-type">cost</cbc:AwardingCriterionTypeCode>
						<cbc:Name languageID="ENG">Nome che si da per aiutare a capire</cbc:Name>
						<cbc:Description languageID="ENG">L'offerta economicamente più vantaggiosa rispetto ai criteri indicati nel capitolato d'oneri, nel bando di gara o di negoziazione o nel documento descrittivo
						</cbc:Description>
					</cac:SubordinateAwardingCriterion>
				</cac:AwardingCriterion>
			</cac:AwardingTerms>
			<cac:TenderRecipientParty>
				<cac:PartyIdentification>
					<cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID>
				</cac:PartyIdentification>
			</cac:TenderRecipientParty>
			<cac:Language>
				<cbc:ID>ENG</cbc:ID>
			</cac:Language>
		</cac:TenderingTerms>
		<cac:TenderingProcess>
			<cac:TenderSubmissionDeadlinePeriod>
				<cbc:EndDate>2023-06-01+02:00</cbc:EndDate>
				<cbc:EndTime>04:56:26+02:00</cbc:EndTime>
			</cac:TenderSubmissionDeadlinePeriod>
			<cac:ParticipationRequestReceptionPeriod>
				<cbc:EndDate>2023-06-02+02:00</cbc:EndDate>
				<cbc:EndTime>01:56:26+02:00</cbc:EndTime>
			</cac:ParticipationRequestReceptionPeriod>
			<cac:AuctionTerms>
				<cbc:AuctionConstraintIndicator>true</cbc:AuctionConstraintIndicator>
			</cac:AuctionTerms>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="framework-agreement">fa-mix</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
		</cac:TenderingProcess>
		<cac:ProcurementProject>
			<cbc:ID>LOT_1_FERNANDA_ID</cbc:ID>
			<cbc:Name languageID="ENG">LOT TITLE</cbc:Name>
			<cbc:Description languageID="ENG">LOT DESCR</cbc:Description>
			<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
			<cac:MainCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03000000</cbc:ItemClassificationCode>
			</cac:MainCommodityClassification>
			<cac:AdditionalCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03113100</cbc:ItemClassificationCode>
			</cac:AdditionalCommodityClassification>
			<cac:ContractExtension>
				<cbc:MaximumNumberNumeric>4</cbc:MaximumNumberNumeric>
			</cac:ContractExtension>
		</cac:ProcurementProject>
	</cac:ProcurementProjectLot>
</ContractNotice>`

**Reply by YvesJo** [reply]
Existence of BT-70-Lot is conditioned by the presence of OPT-060-Lot. One may not exist without the other so they should both be simultaneously present or absent. SO there should be something like:

> <<cac:ContractExecutionRequirement>>
    <cbc:ExecutionRequirementCode listName="conditions">performance</cbc:ExecutionRequirementCode>
    <cbc:Description languageID="ENG">During execution of the contract, ...</cbc:Description>
    <cbc:Description languageID="FRA">Pendant l'exécution du contrat, ...</cbc:Description>
</cac:ContractExecutionRequirement>

or nothing. 
In the provided XML instance we may find OPT-060-Lot:
> <<cac:ContractExecutionRequirement>>
    <cbc:ExecutionRequirementCode listName="conditions">performance</cbc:ExecutionRequirementCode>
</cac:ContractExecutionRequirement>

which then requires the existence of BT-70-Lot

**Reply by rengit62** [reply]
Let's begin from notice 7
I am using sdk 1.7

I **removed** mandatory condition for BT-131(t)-Lot and BT-13111(t)-Lot , and filling both together with the  date, as they where mandatory : 
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/c39cb222-7d29-4109-b472-f54de71c901d)

I got a validation error : 
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/88d62799-c788-4a6b-93ad-c06f41f87ec9)

now I remove date and time for for receipt of tenders ( I can do because I removed mandatory condition in fields,json )
![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/d3bbc496-a45e-4eca-b8b1-dcd8bff25b09)

the notice is CVS valid

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/0138f28b-e3df-41c2-9955-520a69706a66)

The same happens removing time and date for receipt of requests to participate

They are alternative .
Then can you tell me why in fields,json BT-131(t)-Lot and BT-1311(t)-Lot are **both** mandatory ?

"id" : "BT-131(t)-Lot",
    "parentNodeId" : "ND-SubmissionDeadline",
    "name" : "Deadline Receipt Tenders",
    "btId" : "BT-131",
  ************************************************************
  },
    "mandatory" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ **"7",** "8", "9", "16", "17", "18", "19", "20", "21", "22", "23", "24" ],
        "value" : true,
        "severity" : "ERROR"
      } ]


"id" : "BT-1311(t)-Lot",
    "parentNodeId" : "ND-ParticipationRequestPeriod",
    "name" : "Deadline Receipt Requests",
    *******************************************},
    "mandatory" : {
      "value" : false,
      "severity" : "ERROR",
      "constraints" : [ {
        "noticeTypes" : [ **"7"**, "9", "16", "17", "18", "19", "20", "21", "22", "23", "24" ],
        "value" : true,
        "severity" : "ERROR"
      } ]



Bye
Renato

**Reply by YvesJo** [reply, marked as answer]
There are some rules that mandate the existence of one of the deadline and others for the other deadline. It looks like not all cases are covered and for the cases where uncertainty exist, then additional rules may need to be put in place to ensure at least one of the deadline is specified

**Comment by rengit62** [comment]
Hi lab
in eNotices2 https://enotices2.preview.ted.europa.eu/edit-notice/20997?sectionId=GR-Lot&instanceId=LOT-0001&subsectionId=GR-Lot-SubmissionInfo&save=true

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/73732c00-f969-4682-baf9-1cfdb54f0624)

Submission Info seems to have no mandatory field in notice 7
Indeed I just sent a notice that has been submitted without any data/time of submissionInfo, so why do not remove mandatory field in submission info ?
I sent this notice : 
` <?xml version="1.0" encoding="UTF-8"?>
<PriorInformationNotice xmlns="urn:oasis:names:specification:ubl:schema:xsd:PriorInformationNotice-2" xmlns:cac="urn:oasis:names:specification:ubl:schema:xsd:CommonAggregateComponents-2" xmlns:cbc="urn:oasis:names:specification:ubl:schema:xsd:CommonBasicComponents-2" xmlns:efac="http://data.europa.eu/p27/eforms-ubl-extension-aggregate-components/1" xmlns:efbc="http://data.europa.eu/p27/eforms-ubl-extension-basic-components/1" xmlns:efext="http://data.europa.eu/p27/eforms-ubl-extensions/1" xmlns:ext="urn:oasis:names:specification:ubl:schema:xsd:CommonExtensionComponents-2">
	<ext:UBLExtensions>
		<ext:UBLExtension>
			<ext:ExtensionContent>
				<efext:EformsExtension>
					<efac:NoticeSubType>
						<cbc:SubTypeCode listName="notice-subtype">7</cbc:SubTypeCode>
					</efac:NoticeSubType>
					<efac:Organizations>
						<efac:Organization>
							<efbc:GroupLeadIndicator>true</efbc:GroupLeadIndicator>
							<efac:Company>
								<cac:PartyIdentification>
									<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
								</cac:PartyIdentification>
								<cac:PartyName>
									<cbc:Name languageID="ENG">Buyer Orfanizarion</cbc:Name>
								</cac:PartyName>
								<cac:PostalAddress>
									<cbc:CityName>PISA</cbc:CityName>
									<cac:Country>
										<cbc:IdentificationCode listName="country">ITA</cbc:IdentificationCode>
									</cac:Country>
								</cac:PostalAddress>
								<cac:PartyLegalEntity>
									<cbc:CompanyID>111111111111</cbc:CompanyID>
								</cac:PartyLegalEntity>
								<cac:Contact>
									<cbc:Telephone>0512323</cbc:Telephone>
									<cbc:ElectronicMail>ferdi@sds.sds</cbc:ElectronicMail>
								</cac:Contact>
							</efac:Company>
							<efac:TouchPoint>
								<cac:PartyIdentification>
									<cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID>
								</cac:PartyIdentification>
							</efac:TouchPoint>
						</efac:Organization>
					</efac:Organizations>
				</efext:EformsExtension>
			</ext:ExtensionContent>
		</ext:UBLExtension>
	</ext:UBLExtensions>
	<cbc:UBLVersionID>2.3</cbc:UBLVersionID>
	<cbc:CustomizationID>eforms-sdk-1.7</cbc:CustomizationID>
	<cbc:ID schemeName="notice-id">fb803820-dae4-4b67-955d-453f45d683b1</cbc:ID>
	<cbc:IssueDate>2023-06-08+02:00</cbc:IssueDate>
	<cbc:IssueTime>11:15:37+02:00</cbc:IssueTime>
	<cbc:VersionID>02</cbc:VersionID>
	<cbc:RegulatoryDomain>31985R2137</cbc:RegulatoryDomain>
	<cbc:NoticeTypeCode listName="planning">pin-rtl</cbc:NoticeTypeCode>
	<cbc:NoticeLanguageCode listName="language">ENG</cbc:NoticeLanguageCode>
	<cac:ContractingParty>
		<cbc:BuyerProfileURI>http://buyer1.iu</cbc:BuyerProfileURI>
		<cac:ContractingPartyType>
			<cbc:PartyTypeCode listName="buyer-legal-type">body-pl-ra</cbc:PartyTypeCode>
		</cac:ContractingPartyType>
		<cac:ContractingActivity>
			<cbc:ActivityTypeCode listName="authority-activity">education</cbc:ActivityTypeCode>
		</cac:ContractingActivity>
		<cac:Party>
			<cac:PartyIdentification>
				<cbc:ID schemeName="organization">ORG-0001</cbc:ID>
			</cac:PartyIdentification>
		</cac:Party>
	</cac:ContractingParty>
	<cac:TenderingProcess>
		<cbc:ProcedureCode listName="procurement-procedure-type">open</cbc:ProcedureCode>
	</cac:TenderingProcess>
	<cac:ProcurementProject>
		<cbc:Name languageID="ENG">PROC TITLE</cbc:Name>
		<cbc:Description languageID="ENG">PROC DESCR</cbc:Description>
		<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
		<cac:MainCommodityClassification>
			<cbc:ItemClassificationCode listName="cpv">03112000</cbc:ItemClassificationCode>
		</cac:MainCommodityClassification>
	</cac:ProcurementProject>
	<cac:ProcurementProjectLot>
		<cbc:ID schemeName="Lot">LOT-0001</cbc:ID>
		<cac:TenderingTerms>
			<ext:UBLExtensions>
				<ext:UBLExtension>
					<ext:ExtensionContent>
						<efext:EformsExtension>
							<efac:SelectionCriteria>
								<cbc:CriterionTypeCode listName="selection-criterion">ef-stand</cbc:CriterionTypeCode>
								<cbc:Name languageID="ENG">Criterio economico</cbc:Name>
								<cbc:Description languageID="ENG">situazione economica e finanziaria</cbc:Description>
								<cbc:CalculationExpressionCode listName="usage">used</cbc:CalculationExpressionCode>
								<efbc:SecondStageIndicator>true</efbc:SecondStageIndicator>
							</efac:SelectionCriteria>
							<efac:SelectionCriteria>
								<cbc:CriterionTypeCode listName="selection-criterion">sui-act</cbc:CriterionTypeCode>
								<cbc:Name languageID="ENG">criterio professionale</cbc:Name>
								<cbc:Description languageID="ENG">idoneità all'esercizio dell'attività professionale</cbc:Description>
								<cbc:CalculationExpressionCode listName="usage">nyk</cbc:CalculationExpressionCode>
								<efbc:SecondStageIndicator>true</efbc:SecondStageIndicator>
							</efac:SelectionCriteria>
							<efac:SelectionCriteria>
								<cbc:CriterionTypeCode listName="selection-criterion">tp-abil</cbc:CriterionTypeCode>
								<cbc:Name languageID="ENG">Criterio tecnico</cbc:Name>
								<cbc:Description languageID="ENG">Abilita' tecnica e professionale</cbc:Description>
								<efbc:SecondStageIndicator>true</efbc:SecondStageIndicator>
								<efac:CriterionParameter>
									<efbc:ParameterCode listName="number-weight">dec-mid</efbc:ParameterCode>
									<efbc:ParameterNumeric>20</efbc:ParameterNumeric>
								</efac:CriterionParameter>
							</efac:SelectionCriteria>
						</efext:EformsExtension>
					</ext:ExtensionContent>
				</ext:UBLExtension>
			</ext:UBLExtensions>
			<cbc:VariantConstraintCode listName="permission">allowed</cbc:VariantConstraintCode>
			<cbc:FundingProgramCode listName="eu-funded">eu-funds</cbc:FundingProgramCode>
			<cbc:RequiredCurriculaCode listName="requirement-stage">par-requ</cbc:RequiredCurriculaCode>
			<cbc:LatestSecurityClearanceDate>2023-05-25+02:00</cbc:LatestSecurityClearanceDate>
			<cac:CallForTendersDocumentReference>
				<cbc:ID>FERDI</cbc:ID>
				<cbc:DocumentType>restricted-document</cbc:DocumentType>
				<cac:Attachment>
					<cac:ExternalReference>
						<cbc:URI>http://sdsd.dfd</cbc:URI>
					</cac:ExternalReference>
				</cac:Attachment>
			</cac:CallForTendersDocumentReference>
			<cac:TendererQualificationRequest>
				<cac:SpecificTendererRequirement>
					<cbc:TendererRequirementTypeCode listName="reserved-procurement">none</cbc:TendererRequirementTypeCode>
				</cac:SpecificTendererRequirement>
			</cac:TendererQualificationRequest>
			<cac:ContractExecutionRequirement>
				<cbc:Description languageID="ENG">E TUTTI VISSERO FELICI E CONTENTI</cbc:Description>
			</cac:ContractExecutionRequirement>
			<cac:ContractExecutionRequirement>
				<cbc:ExecutionRequirementCode listName="reserved-execution">yes</cbc:ExecutionRequirementCode>
			</cac:ContractExecutionRequirement>
			<cac:AdditionalInformationParty>
				<cac:PartyIdentification>
					<cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID>
				</cac:PartyIdentification>
			</cac:AdditionalInformationParty>
			<cac:AppealTerms>
				<cac:AppealReceiverParty>
					<cac:PartyIdentification>
						<cbc:ID schemeName="touchpoint">TPO-0001</cbc:ID>
					</cac:PartyIdentification>
				</cac:AppealReceiverParty>
			</cac:AppealTerms>
			<cac:Language>
				<cbc:ID>ENG</cbc:ID>
			</cac:Language>
			<cac:SecurityClearanceTerm>
				<cbc:Code listName="required">true</cbc:Code>
				<cbc:Description languageID="ENG">O si lavora in sicurezza oppure si fa sciopero !</cbc:Description>
			</cac:SecurityClearanceTerm>
		</cac:TenderingTerms>
		<cac:TenderingProcess>
			<cbc:GovernmentAgreementConstraintIndicator>true</cbc:GovernmentAgreementConstraintIndicator>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="framework-agreement">fa-mix</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
			<cac:ContractingSystem>
				<cbc:ContractingSystemTypeCode listName="dps-usage">dps-nlist</cbc:ContractingSystemTypeCode>
			</cac:ContractingSystem>
		</cac:TenderingProcess>
		<cac:ProcurementProject>
			<cbc:ID>INTERNAL_MARY_IDENTIFIER</cbc:ID>
			<cbc:Name languageID="ENG">LOT TITLE</cbc:Name>
			<cbc:Description languageID="ENG">LOT DESCR</cbc:Description>
			<cbc:ProcurementTypeCode listName="contract-nature">supplies</cbc:ProcurementTypeCode>
			<cac:MainCommodityClassification>
				<cbc:ItemClassificationCode listName="cpv">03113100</cbc:ItemClassificationCode>
			</cac:MainCommodityClassification>
		</cac:ProcurementProject>
	</cac:ProcurementProjectLot>
</PriorInformationNotice>`
Bye
Renato
